### PR TITLE
better use of void

### PR DESCRIPTION
**Pull request by Borda**
## What does this PR do?

this shall fix all mypy in deprecated metrics

## Before submitting
- [ ] Was this discussed/approved via a GitHub issue? (not for typos and docs)
- [x] Did you read the [contributor guideline](https://github.com/PyTorchLightning/pytorch-lightning/blob/master/.github/CONTRIBUTING.md), **Pull Request** section?
- [x] Did you make sure your PR does only one thing, instead of bundling different changes together?
- [x] Did you make sure to update the documentation with your changes? (if necessary)
- [x] Did you write any new necessary tests? (not for typos and docs)
- [ ] Did you verify new and existing tests pass locally with your changes?
- [x] Did you update the [CHANGELOG](https://github.com/PyTorchLightning/pytorch-lightning/blob/master/CHANGELOG.md)? (not for typos, docs, test updates, or internal minor changes/refactorings)

## PR review
Anyone in the community is free to review the PR once the tests have passed.
Before you start reviewing make sure you have read [Review guidelines](https://github.com/PyTorchLightning/pytorch-lightning/wiki/Review-guidelines). In short, see the following bullet-list:

 - [x] Is this pull request ready for review? (if not, please submit in draft mode)
 - [x] Check that all items from **Before submitting** are resolved
 - [x] Make sure the title is self-explanatory and the description concisely explains the PR
 - [x] Add labels and milestones (and optionally projects) to the PR so it can be classified

## Did you have fun?
Make sure you had fun coding 🙃

### CODE REVIEW

File: pytorch_lightning/metrics/classification/accuracy.py [hunk 1]
```diff
@@ -37,4 +38,4 @@ def __init__(
         .. deprecated::
             Use :class:`~torchmetrics.Accuracy`. Will be removed in v1.5.0.
         """
-        _ = threshold, top_k, subset_accuracy, compute_on_step, dist_sync_on_step, process_group, dist_sync_fn
+        void(threshold, top_k, subset_accuracy, compute_on_step, dist_sync_on_step, process_group, dist_sync_fn)
```
Comment (carmocca) [inline pytorch_lightning/metrics/classification/accuracy.py:41]: Why do we even have this line?

The docstring acts as `pass`

**Review by ethanwharris — APPROVED** [review]

**Review by awaelchli — APPROVED** [review]

File: pytorch_lightning/metrics/classification/accuracy.py (hunk 1 above)
Comment (Borda) [reply pytorch_lightning/metrics/classification/accuracy.py:41]: yes, but then IDE complains that you have unused arguments...

Comment (carmocca) [reply pytorch_lightning/metrics/classification/accuracy.py:41]: Does it? I don't get any complain for the stubs in `lightning.py`, for example

Screenshot 2021-06-02 at 21 04 07

Comment (Borda) [reply pytorch_lightning/metrics/classification/accuracy.py:41]: not sure what you mean here...

File: pytorch_lightning/metrics/classification/accuracy.py [hunk 2]
```diff
@@ -13,6 +13,7 @@
 # limitations under the License.
 from typing import Any, Callable, Optional
 
+from deprecate import void
```
Comment (justusschock) [inline pytorch_lightning/metrics/classification/accuracy.py:16]: Why does this come from your deprecate package @Borda ? This doesn't look like a good fit for me. Can we move it to our trips here since it's not about depreciation?

Comment (Borda) [reply pytorch_lightning/metrics/classification/accuracy.py]: what do you mean by trips? moved to our utils...

Comment (justusschock) [reply pytorch_lightning/metrics/classification/accuracy.py]: That's what I meant. Was autocorrect, sorry :D I really need to turn off the automatic changes

File: pytorch_lightning/metrics/utils.py [hunk 3]
```diff
@@ -15,7 +15,7 @@
 from typing import Optional
 
 import torch
-from deprecate import deprecated
+from deprecate import deprecated, void
```
Comment (justusschock) [inline pytorch_lightning/metrics/utils.py:18]: That's not really better here :D You only forwarded the import, but I think this void thing shouldn't live in the deprecate package in the first place, since this has nothing to do with deprecation.

**Review by justusschock — APPROVED** [review]
Just one more comment about its origin. It's fine for me although the origin isn't intuitive for me :)

File: pytorch_lightning/metrics/utils.py (hunk 3 above)
Comment (Borda) [reply pytorch_lightning/metrics/utils.py:18]: ok, copy-pasted the implementation :]

File: pytorch_lightning/metrics/classification/accuracy.py (hunk 1 above)
Comment (carmocca) [reply pytorch_lightning/metrics/classification/accuracy.py:40]: You said the IDE complains about unused arguments, IIRC you also use PyCharm.

Can you elaborate on those complaints? Where are they?

My screenshot is an example of another function where the arguments are "unused" and the linter doesn't seem to show any warnings.

Comment (Borda) [reply pytorch_lightning/metrics/classification/accuracy.py:40]: yes, but in your case you have it in docs at least...
one of my cases was running code inspector and there is a group called unused arguments
could be also https://pycodequ.al/docs/pylint-messages/w0613-unused-argument.html ?

Comment (carmocca) [reply pytorch_lightning/metrics/classification/accuracy.py:40]: then why not use `# noqa: W0613` or `# pylint: disable=unused-argument`

instead of introducing `void`

**Review by tchaton — COMMENTED** [review]
Not sure about this one.

**Comment by Borda** [comment]
> it doesn't seem intuitive for new readers.

what do you mean?
